File: FDev/WebViewActivity/app/src/main/java/com/example/webviewactivity/MainActivity.java
package com.example.webviewactivity;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.webkit.WebSettings;
import android.webkit.WebView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        WebView webView = (WebView) findViewById(R.id.webview);
        WebSettings webSettings = webView.getSettings();
        webSettings.setJavaScriptEnabled(true);
        webView.loadUrl("https://www.hlb.com.my:443/en/personal-banking/promotions/cards-rm5-cashback-with-hlb-cards.html?ecp\\u003dCACNMAHLCPFSD011118L0");
    }
}

File: FDev/DaggerDemo/app/src/main/java/com/example/daggerdemo/MainActivity.kt
package com.example.daggerdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.daggerdemo.R
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject
     lateinit var smartPhone:SmartPhone

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        val smartPhone = SmartPhone(Battery(),SIMCard (ServiceProvider()),MemoryCard()).makeACallWithRecording()

//        smartPhone = DaggerDgComponent.create().getSmartPhoneProvider()
//         DaggerDgComponent.create().inject(this)

        ( application as SmartPhoneApplication).dgComponent.inject(this)

        smartPhone?.makeACallWithRecording()

    }
}
